### ISSUE TITLE
default and initial of a property should be per instance and not per class

### ISSUE THREAD

**Issue by codeStryke** [Netflix/x-element#79]
Opening up an issue to discuss why `default` and `initial` values of a `property` are set per class and not per instance.
@theengineear @klebba

**Comment by klebba** [comment]
@codeStryke I'm not sure I follow, you would have to propose an alternative approach. How would it work to configure these per instance at the class definition of the element?

You can override the class level definitions on your instances like this:

```
...
// property definition in class
{
  myNeatProperty: {
    type: String,
    initial: 1,
  },
},

// html
<my-cool-element my-neat-property="4"></my-cool-element>

// javascript
const element = document.querySelector('my-cool-element');
element.myNeatProperty = 4;
```

**Comment by codeStryke** [comment]
I should have clarified, when the `default` or `initial` value is set via a function, then, it should be per instance and not at the class level.

**Comment by reliable-casey** [comment]
Ah ok, understood; that sounds like a bug. Thanks for reporting!

**Comment by theengineear** [comment]
Man, I'm really losing it today — forgot to close this from my PR... This is closed by #81.
